### ISSUE TITLE
golint doesn't return non-zero when it finds lint

### ISSUE THREAD

**Issue by tsuna** [golang/lint#65]
Trying to make my build fail when golint finds lint, but that's kind of annoying to do right now with golint because it always exits 0.

**Comment by phylake** [comment]
I'd like a useful exit code as well - for my git pre-commit hook

**Comment by dsymonds** [comment]
Don't use golint in a git pre-commit hook. You'll be awfully grumpy when it starts finding false positives in existing code.

**Comment by phylake** [comment]
Actually I'm grumpy that your response to 2 requests for a non-zero exit code is "don't do the thing you're trying to do". How about this: do you have a good reason _not_ to add a non-zero exit code?

**Comment by daemonza** [comment]
would love to have a exit 1 as well

**Comment by jszwedko** [comment]
@dsymonds while I agree with you that you shouldn't have it failing your builds, it is still nice to have the option to use it as a commit hook or something similar. It only makes the tool more flexible, I'm not sure what downsides there are.

**Comment by zhtheyun** [comment]
+1 for exit 1 as well.

**Comment by gdm85** [comment]
@dsymonds if this is a CLI tool (command-line-interface) it should set an exit code on error, as all other CLI tools do and - last but not least - as user would expect.

**Comment by jszwedko** [comment]
I agree with the philosophy of not returning non-zero by default as long as the tool did not actually fail to run (as opposed to it ran and found errors), but I think being able to configure this behavior (like `curl --fail` does) would be very useful.

**Comment by adrianmacneil** [comment]
I would also like to return a non-zero exit code. I also think this is a pretty reasonable default - if people want to specifically allow failures in a script this is pretty easy to work around.

Seems strange to me that go's philosophy is all so "do everything exactly our way or it will immediately fail with an error", yet the lint tool's philosophy is "everything is just a suggestion, you shouldn't be strict about this".

**Comment by hit9** [comment]
+1 for exit 1. We can use `golint || true` if the lint errors are really not cared.

**Comment by kostya-sh** [comment]
if you care about lint errors you can use `test -z "$(golint)"`

**Comment by hit9** [comment]
cool！

**Comment by n8gard** [comment]
I also am in the camp that would love to fail a CI run when lint is found. Thanks for the tip on `fgt`.

**Comment by tj** [comment]
It's completely subjective, it should absolutely fail builds for an open-source project that has such requirements. +1 for proper exit code, it's simple enough to just ignore it if you don't want it.

**Comment by marcuswestin** [comment]
What @tj said. +1. Am forced to an annoying workaround instead.

**Comment by marcuswestin** [comment]
An option would be to add a flag to fail when any lint is detected.

**Comment by marcuswestin** [comment]
@tj: This is what I ended using for a lint- and vet-safe `make test`:

```
test: lint vet
	go test -v .

test-race: test
	go test --race -v .

lint:
	golint .
	test -z "$$(golint .)"

vet:
	go vet .
```

**Comment by tochiai** [comment]
+1, would also like an exit 1 or a flag as @marcuswestin suggested

**Comment by danilobuerger** [comment]
+1

**Comment by rsc** [comment]
This is clearly a feature that is important to a significant class of users. I don't see why this is "wontfix". It's true that we can't change the default behavior, because that might break current scripts as mentioned above, but we can certainly add a flag, say -s (for status) that makes golint set the exit status depending on whether any problems are found.

**Comment by dsymonds** [comment]
Resolved by 58f662d.
